### PR TITLE
press summary email & form

### PR DESCRIPTION
**Pull request by dreamfall**

### CODE REVIEW

File: app/controllers/users/press_summaries_controller.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+class Users::PressSummariesController < Users::BaseController
+  before_action :check_deadline
+
+  expose(:form_answer) do
+    current_user.account.
+                form_answers.
```
Comment (matthewford) [inline app/controllers/users/press_summaries_controller.rb]: Place the . on the next line, together with the method name.

File: app/controllers/users/press_summaries_controller.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+class Users::PressSummariesController < Users::BaseController
+  before_action :check_deadline
+
+  expose(:form_answer) do
+    current_user.account.
```
Comment (matthewford) [inline app/controllers/users/press_summaries_controller.rb]: Place the . on the next line, together with the method name.

File: app/controllers/users/press_summaries_controller.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                form_answers.
+                find(params[:form_answer_id])
+  end
+
+  def show
+    @press_summary = form_answer.press_summary
+    @press_summary.first_name   ||= current_user.first_name
```
Comment (matthewford) [inline app/controllers/users/press_summaries_controller.rb]: Unnecessary spacing detected.

File: app/controllers/users/press_summaries_controller.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    @press_summary.last_name    ||= current_user.last_name
```
Comment (matthewford) [inline app/controllers/users/press_summaries_controller.rb]: Unnecessary spacing detected.

File: app/controllers/users/press_summaries_controller.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    @press_summary.email        ||= current_user.email
```
Comment (matthewford) [inline app/controllers/users/press_summaries_controller.rb]: Unnecessary spacing detected.

File: app/renderers/mail_renderer.rb [hunk 6]
```diff
@@ -10,7 +10,7 @@ def default_url_options
   end
 
   # this will be removed after all methods are implemented
-  %w(reminder_to_submit ep_reminder_support_letters winners_notification winners_reminder_to_submit winners_press_release_comments_request unsuccessfull_notification all_unsuccessfull_feedback).each do |method|
+  %w(reminder_to_submit ep_reminder_support_letters winners_notification winners_reminder_to_submit unsuccessfull_notification all_unsuccessfull_feedback).each do |method|
```
Comment (matthewford) [inline app/renderers/mail_renderer.rb]: Line is too long. [171/100]

File: app/services/notifiers/email_notification_service.rb [hunk 7]
```diff
@@ -18,7 +18,7 @@ def run
   end
 
   # this will be removed after all methods are implemented
-  %w(reminder_to_submit ep_reminder_support_letters winners_reminder_to_submit winners_press_release_comments_request unsuccessfull_notification all_unsuccessfull_feedback).each do |method|
+  %w(reminder_to_submit ep_reminder_support_letters winners_reminder_to_submit unsuccessfull_notification all_unsuccessfull_feedback).each do |method|
```
Comment (matthewford) [inline app/services/notifiers/email_notification_service.rb]: Line is too long. [150/100]

File: spec/mailers/users/winners_press_release_spec.rb [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+    it "renders the headers" do
+      expect(mail.subject).to eq("[Queen's Awards for Enterprise] Press Comment")
+      expect(mail.to).to eq([user.email])
+      expect(mail.from).to eq(["info@queensawards.org.uk"])
+    end
+
+    it "renders the body" do
+      expect(mail.body.encoded).to match(user.decorate.full_name)
+      expect(mail.body.encoded).to have_link("Review Press Comment", href: users_form_answer_press_summary_url(form_answer))
```
Comment (matthewford) [inline spec/mailers/users/winners_press_release_spec.rb]: Line is too long. [124/100]

File: spec/mailers/users/winners_press_release_spec.rb [hunk 9]
```diff
@@ -0,0 +1,38 @@
+require "rails_helper"
+
+describe Users::WinnersPressRelease do
+  let!(:user) do
+    FactoryGirl.create :user, :completed_profile,
+                              first_name: "Admin John",
```
Comment (matthewford) [inline spec/mailers/users/winners_press_release_spec.rb]: Align the parameters of a method call if they span more than one line.

File: spec/mailers/users/winners_press_release_spec.rb [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+                              role: "account_admin"
+  end
+
+  describe "#notify" do
+    let(:form_answer) do
+      FactoryGirl.create :form_answer, :submitted, :innovation,
+                                                   user: user,
```
Comment (matthewford) [inline spec/mailers/users/winners_press_release_spec.rb]: Align the parameters of a method call if they span more than one line.

File: app/renderers/mail_renderer.rb [hunk 11]
```diff
@@ -56,4 +63,8 @@ def render(assigns, template)
   def dummy_user(first_name, last_name)
     User.new(first_name: first_name, last_name: last_name).decorate
   end
+
+  def form_answer
+    FormAnswer.new(id: 0, award_type: "promotion").decorate
```
Comment (jaleszek) [inline app/renderers/mail_renderer.rb]: what id = 0 does mean here ?

Comment (dreamfall) [inline app/renderers/mail_renderer.rb]: ID is needed to build an URL on settings page. But we don't wanna use real ID for the example url.

**Comment by jaleszek** [comment]
looks good :+1:

File: app/renderers/mail_renderer.rb (hunk 11 above)
Comment (matthewford) [inline app/renderers/mail_renderer.rb]: perhaps use nil?

Comment (dreamfall) [inline app/renderers/mail_renderer.rb]: @matthewford well, url helpers throw ActionController::UrlGenerationError if ID is nil.

Comment (matthewford) [inline app/renderers/mail_renderer.rb]: touché

**Pull request merged**